cirq/value/duration.py

```diff
@@ -13,120 +13,234 @@
-            return cls(picos=duration_in_picos)
+        return Duration(duration)
 
-        raise TypeError(
-            'Only datetime.timedelta and cirq.Duration are supported.')
+    def _is_parameterized_(self):
+        from cirq import protocols
+        return protocols.is_parameterized(self._picos)
 
-    def total_picos(self) -> float:
+    def _resolve_parameters_(self, param_resolver):
+        from cirq import protocols
+        return Duration(
+            picos=protocols.resolve_parameters(self._picos, param_resolver))
+
+    def total_picos(self) -> Union[sympy.Basic, float]:
         """Returns the number of picoseconds that the duration spans."""
         return self._picos
 
-    def total_nanos(self) -> float:
+    def total_nanos(self) -> Union[sympy.Basic, float]:
         """Returns the number of nanoseconds that the duration spans."""
-        return self._picos / 1000.0
+        return self._picos / 1000
+
+    def total_micros(self) -> Union[sympy.Basic, float]:
+        """Returns the number of microseconds that the duration spans."""
+        return self._picos / 1000_000
+
+    def total_millis(self) -> Union[sympy.Basic, float]:
+        """Returns the number of milliseconds that the duration spans."""
+        return self._picos / 1000_000_000
 
     def __add__(self, other) -> 'Duration':
-        if not isinstance(other, (type(self), timedelta)):
+        other = _attempt_duration_like_to_duration(other)
+        if other is None:
             return NotImplemented
-        other = Duration.create(other)
         return Duration(picos=self._picos + other._picos)
 
     def __radd__(self, other) -> 'Duration':
         return self.__add__(other)
 
     def __sub__(self, other) -> 'Duration':
-        other = Duration.create(other)
+        other = _attempt_duration_like_to_duration(other)
+        if other is None:
+            return NotImplemented
         return Duration(picos=self._picos - other._picos)
 
     def __rsub__(self, other) -> 'Duration':
-        other = Duration.create(other)
+        other = _attempt_duration_like_to_duration(other)
+        if other is None:
+            return NotImplemented
         return Duration(picos=other._picos - self._picos)
 
     def __mul__(self, other) -> 'Duration':
-        if not isinstance(other, (int, float)):
+        if not isinstance(other, (int, float, sympy.Basic)):
             return NotImplemented
         return Duration(picos=self._picos * other)
 
     def __rmul__(self, other) -> 'Duration':
         return self.__mul__(other)
 
     def __truediv__(self, other) -> Union['Duration', float]:
-        if isinstance(other, (int, float)):
+        if isinstance(other, (int, float, sympy.Basic)):
             return Duration(picos=self._picos / other)
-        if isinstance(other, type(self)):
-            return self._picos / other._picos
+
+        other_duration = _attempt_duration_like_to_duration(other)
```

Comment:
If div can produce a Duration or a Float should mul do the same ?

Reply:
Nope. second/second is a scalar, but second*second is neither a scalar nor a duration.